typed-racket-lib/typed-racket/types/pairwise-intersect.rkt

```diff
@@ -0,0 +1,121 @@
+#lang racket
+
+;; This module provides a function pairwise-intersect for calculating a lower
+;; bound of two types with respect to the precision order (*not* the subtype
+;; order). Contract typechecking uses this function to determine the type of a
+;; contracted value, given the value's type and the contract's output type.
+
+(require "../utils/utils.rkt"
+         (rep type-rep prop-rep object-rep values-rep)
+         (types subtype)
+         (only-in (infer infer) intersect)
+         racket/match)
+
+(provide pairwise-intersect)
+
+(define (pairwise-intersect/arr s t)
+  (match* (s t)
```

Comment:
We've moved away from `arr` as an abbreviation in TR's implementation due to some common ambiguities with what it stands for/implies/etc.

Would `pairwise-intersect/arrow` work?